torch/_inductor/ir.py

```diff
@@ -4706,27 +4731,36 @@ def require_stride_order(cls, x, order, allow_padding=False):
                     if is_stride_order_storage_and_layout(x, order)
                     else order,
                     allow_padding=allow_padding,
+                    actual_strides=actual_strides,
                 )
                 return x
-            elif isinstance(
-                x.get_layout(), FixedLayout
-            ) and x.get_layout().is_stride_ordered(order):
+            elif isinstance(x.get_layout(), FixedLayout) and (
+                (order and x.get_layout().is_stride_ordered(order))
+                or (actual_strides and x.get_layout().stride == actual_strides)
```

Comment:
This is wrong. can't simply check `actual_strides and x.get_layout().stride == actual_strides`, or it will change the reinterpretview's implementation here.

Reply:
Need to find another way to fix the bug when we set channel_last.

Reply:
`x.get_layout().stride` or `actual_strides ` can be a tuple or list, so need to convert them to same type for comparison.

Reply:
> This is wrong. can't simply check `actual_strides and x.get_layout().stride == actual_strides`, or it will change the reinterpretview's implementation here.

I added a check to see if ReinterpretView involved outside this function, and skip it.